lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1135,85 +1054,49 @@ export class UserAgentApplication {
+        const popUpWithHash = WindowUtils.getPopUpWithHash(locationHash);
+        const isPopupOrIframe = !!(iframeWithHash || popUpWithHash);
 
         // if (window.parent !== window), by using self, window.parent becomes equal to window in getResponseState method specifically
-        const stateInfo = self.getResponseState(hash);
+        const stateInfo = this.getResponseState(locationHash);
 
         let tokenResponseCallback: (response: AuthResponse, error: AuthError) => void = null;
 
-        self.logger.info("Returned from redirect url");
+        this.logger.info("Returned from redirect url");
         // If parent window is the msal instance which opened the current window (iframe)
-        if (this.parentIsMsal()) {
-            tokenResponseCallback = window.parent.callbackMappedToRenewStates[stateInfo.state];
-        }
-        // Current window is window opener (popup)
-        else if (isWindowOpenerMsal) {
-            tokenResponseCallback = window.opener.callbackMappedToRenewStates[stateInfo.state];
-        }
-        // Redirect cases
-        else {
+        if (isPopupOrIframe) {
+            tokenResponseCallback = window.callbackMappedToRenewStates[stateInfo.state];
+        } else {
+            // Redirect cases
             tokenResponseCallback = null;
             // if set to navigate to loginRequest page post login
-            if (self.config.auth.navigateToLoginRequestUrl) {
-                self.cacheStorage.setItem(Constants.urlHash, hash);
-                if (window.parent === window && !isPopup) {
-                    window.location.href = self.cacheStorage.getItem(Constants.loginRequest, self.inCookie);
+            if (this.config.auth.navigateToLoginRequestUrl) {
+                this.cacheStorage.setItem(Constants.urlHash, locationHash);
+                if (window.parent === window) {
+                    window.location.href = this.cacheStorage.getItem(Constants.loginRequest, this.inCookie);
                 }
                 return;
             }
-            else {
-                window.location.hash = "";
-            }
+
             if (!this.redirectCallbacksSet) {
                 // We reached this point too early - cache hash, return and process in handleRedirectCallbacks
-                self.cacheStorage.setItem(Constants.urlHash, hash);
+                this.cacheStorage.setItem(Constants.urlHash, locationHash);
                 return;
             }
         }
 
-        self.processCallBack(hash, stateInfo, tokenResponseCallback);
+        this.processCallBack(locationHash, stateInfo, tokenResponseCallback);
 
         // If current window is opener, close all windows
-        if (isWindowOpenerMsal) {
-            for (let i = 0; i < window.opener.openedWindows.length; i++) {
-                window.opener.openedWindows[i].close();
-            }
+        if (isPopupOrIframe) {
+            WindowUtils.closePopups();
```

Comment:
So is it decided not to close iframes in this PR?

Reply:
For now, no.